source/auth/auth.rst

```diff
@@ -971,6 +971,12 @@ gssapiServiceName (deprecated)
 	An alias for ``authMechanismProperties=SERVICE_NAME:mongodb``.
 
 
+Errors
+------
+
+Drivers SHOULD raise an error if the ``authSource`` option is specified in the connection string with an empty value, e.g. ``mongodb://localhost/admin?authSource=``.
```

Comment:
I don't think it conflicts with SPEC-1511. https://github.com/mongodb/specifications/commit/af6b8b14606d43f1938a2a8da23801f05c6cf4ab doesn't care about the _value_ of `authSource` -- it just states that the presence of `authSource` alone is not sufficient to invoke authentication.

This issue is prohibiting an empty value for `authSource` in _any_ situation -- and if an error is raised (MUST per your other suggestion, which I support) then the driver would never even make it to the point where it might consider invoking auth.

Reply:
Let's at least add a test that asserts that `mongodb://localhost/admin?authSource=` is also invalid.

Reply:
I have added a test with the suggested connection string.